pkg/controller/common.go

```diff
@@ -1529,29 +1527,44 @@ func (c *SharedState) createReplicas(ctx context.Context, remainingReplicas int,
 		remainingReplicas--
 	}
 
+	for _, podObj := range podObjs {
+		azr.getSharedState().eventRecorder.Eventf(podObj.DeepCopyObject(), v1.EventTypeNormal, ReplicaAttachmentCreationSuccessEvent, "%d replica attachment(s) created for volume %s", requiredReplicas-remainingReplicas, volumeName)
+	}
+
 	if remainingReplicas > 0 {
 		//no failed replica attachments, but there are still more replicas to reach MaxShares
 		request := ReplicaRequest{VolumeName: volumeName, Priority: remainingReplicas}
 		c.priorityReplicaRequestsQueue.Push(&request)
+		for _, podObj := range podObjs {
+			azr.getSharedState().eventRecorder.Eventf(podObj.DeepCopyObject(), v1.EventTypeWarning, ReplicaAttachmentCreationFailedEvent, "%d/%d replica attachment(s) for volume %s could not be created. Not enough suitable nodes", remainingReplicas, requiredReplicas, volumeName)
+		}
 	}
 	return nil
 }
 
-func (c *SharedState) getNodesForReplica(ctx context.Context, volumeName string, pods []string, numReplica int, azr azReconciler) ([]string, error) {
+func (c *SharedState) getNodesForReplica(ctx context.Context, volumeName string, podObjs []v1.Pod, numReplica int, azr azReconciler) ([]string, error) {
 	var err error
-	if pods == nil {
-		pods, err = c.getPodsFromVolume(volumeName)
+	if len(podObjs) == 0 {
+		pods, err := c.getPodsFromVolume(volumeName)
```

Comment:
Perhaps `getPodsFromVolume` should return `[]v1.Pod` instead of just pod names since in both usages we're now getting the `v1.Pod` anyway?

Reply:
Done